vega_sim/scenario/network/scenario.py

```diff
@@ -0,0 +1,136 @@
+    ):
+
+        dotenv.load_dotenv()
+        self.wallet_name = os.environ.get("VEGA_USER_WALLET_NAME")
+        self.wallet_pass = os.environ.get("VEGA_USER_WALLET_PASS")
+
+        self.num_steps = num_steps
+        self.step_length_seconds = step_length_seconds
+
+        self.market_name = market_name
+        self.market_id = None
+        self.asset_name = None
+
+        self.random_state = np.random.RandomState()
+
+    def set_up_background_market(
+        self,
+        vega: VegaServiceNetwork,
+        tag: str = "",
+        random_state: Optional[np.random.RandomState] = None,
+    ) -> NetworkEnvironment:
+
+        self.market_id = [
+            m.id
+            for m in vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+        market_info = vega.market_info(market_id=self.market_id)
+        self.asset_name = (
+            market_info.tradable_instrument.instrument.future.settlement_asset
+        )
+
+        mo_trader = MarketOrderTrader(
+            wallet_name=self.wallet_name,
+            wallet_pass=self.wallet_pass,
+            market_name=self.market_name,
+            asset_name=self.asset_name,
+            tag=str(tag),
+            buy_intensity=200,
+            sell_intensity=200,
+            base_order_size=0.01,
+            random_state=random_state,
+        )
+
+        lo_trader = LimitOrderTrader(
+            wallet_name=self.wallet_name,
+            wallet_pass=self.wallet_pass,
+            market_name=self.market_name,
+            asset_name=self.asset_name,
+            tag=str(tag),
+            random_state=random_state,
+            side_opts={"SIDE_BUY": 0.5, "SIDE_SELL": 0.5},
+            time_in_force_opts={"TIME_IN_FORCE_GTC": 1.0},
+            buy_intensity=200,
+            sell_intensity=200,
+            buy_volume=0.01,
+            sell_volume=0.01,
+            mean=-5,
+            sigma=0.5,
+        )
+
+        env = NetworkEnvironment(
+            agents=[
+                mo_trader,
+                lo_trader,
+            ],
+            n_steps=self.num_steps,
+            vega_service=vega,
+            step_length_seconds=self.step_length_seconds,
+        )
+
+        return env
+
+    def run_iteration(
+        self,
+        vega: VegaServiceNetwork,
+        random_state: Optional[np.random.RandomState] = None,
+    ):
+        env = self.set_up_background_market(
+            vega=vega, tag="", random_state=random_state
+        )
+        result = env.run()
+        return result
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--network")
+    parser.add_argument("--debug", action="store_true")
+
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO if not args.debug else logging.DEBUG)
+
+    scenario = Fairground(
+        num_steps=200,
+        step_length_seconds=1,
+        market_name="UNIDAI Monthly (30 Jun 2022)",
```

Comment:
In a future PR we should make this all configurable so someone can set this scenario up to just run this and trigger a bunch of customisable bots

Reply:
Sounds good, I'll raise a PR early next week for this. Out of interest do you think scenarios running on Vega networks should be kept separate from scenarios run on nullchain? Or should we modify adhoc.py to be capable of running scenarios on both services?

Reply:
I guess the later would be nice because it would let us test a service on nullchain and then deploy it to a network with no changes.